spec/defines/clientry_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'spec_helper_puppet'
+
+describe 'jboss::clientry', :type => :define do
+  shared_examples 'completly working define' do
+    it { is_expected.to compile }
+    it { is_expected.to contain_jboss_confignode(title).with ({
+      :ensure => 'present',
+      :path   => 'profile/test'
+      }) }
+    it { is_expected.to contain_jboss_confignode(title).that_requires('Anchor[jboss::package::end]')}
+    it { is_expected.to contain_class 'jboss' }
+    it { is_expected.to contain_jboss__clientry(title) }
```

Comment:
!fix describe parameters of this clientry, especially that you are testing this type!